src/utils/helpers/structure/updateSectionsAndPages.ts

```diff
@@ -0,0 +1,25 @@
+import { FullStructure } from '../../types'
+
+const updateSectionsAndPages = (newStructure: FullStructure): FullStructure => {
+  const sortedSections = Object.values(newStructure.sections).sort(
+    (sectionOne, sectionTwo) => sectionOne.details.index - sectionTwo.details.index
+  )
+  const sortedPages = sortedSections
+    .map((section) =>
+      Object.values(section.pages).sort((pageOne, pageTwo) => pageOne.number - pageTwo.number)
+    )
+    .flat()
+
+  // Set if section is or not active (in case no element is visible & active - should be inactive/hidden)
+  sortedSections.forEach((section) => {
+    section.details.active = Object.values(section.pages).some((page) =>
+      Object.values(page.state).some(
+        ({ element: { isVisible, isEditable } }) => isVisible && isEditable
```

Comment:
I think it should be just `isVisible`, not `isEditable`. Cos if there's uneditable elements, we would still want to see them.

Reply:
No problem - we still can't review it, so I think it's ok.

I'll also remember to remove the visibility from static fields in this case...